modules/README.md

```diff
@@ -141,9 +145,123 @@ interface ISafeProtocol712SignatureValidator {
         bytes32 domainSeparator,
         bytes32 typeHash,
```

Comment:
What is the advantage of having `typeHash` and `encodedData` via just a `messageHash`?

Reply:
Also can a `ISafeProtocol712SignatureValidator` assume that the caller has checked that `_hash` is the 712 hash of the other params?

Reply:
> What is the advantage of having `typeHash` and `encodedData` via just a `messageHash`?

A signature validator as per current spec is specific only to a domain. It should ideally support any type of signed message. If typehash is hardcoded in the validator contract then, hash validation would be specific to domain and type. I think it would be too restrictive in that case.

So, by expecting a `typeHash` and encoded message as a validation parameter, signature validator can validate hash for any typed data

Reply:
> Also can a `ISafeProtocol712SignatureValidator` assume that the caller has checked that `_hash` is the 712 hash of the other params?

I propose that we do it in Signature validator manager. Added in spec (sequence diagram).

Reply:
removed typeHash and encodeData